### PR TITLE
Limit the rssChangeLog to 20 and read system property for customization

### PR DESCRIPTION
**Pull request by StefanSpieker**
Limit the rssChangelog to 20 entries, as suggested by
See [JENKINS-18992](https://issues.jenkins.io/browse/JENKINS-18992).

This superseeds the PR #4963 

### Proposed changelog entries

* Limit the Rss Changelog to 20 and make it customizable through the new property `hudson.model.Job.rssChangeLogLimit`

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by basil — COMMENTED** [review]
A limit on the number of entries strikes me as reasonable, but a system property to control the limit seems like a poor user experience. I think the standard paradigm for things like this is to employ pagination, with a query string parameter to govern which page of the results to retrieve. See [RFC 5005, Feed Paging and Archiving, §3](https://www.rfc-editor.org/rfc/rfc5005#section-3).

File: core/src/main/java/hudson/model/Job.java [hunk 1]
```diff
@@ -135,6 +136,8 @@
 
     private static final Logger LOGGER = Logger.getLogger(Job.class.getName());
 
+    private static /* not final */ int RSS_LIMIT = SystemProperties.getInteger(Job.class.getName() + ".rssChangeLogLimit", 20);
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/Job.java:139]:
```suggestion
    private static /* not final */ int DEFAULT_RSS_LIMIT = SystemProperties.getInteger(Job.class.getName() + ".rssChangeLogLimit", 20);
```

**Review by oleg-nenashev — COMMENTED** [review]
I suggest adding an HTTM argument which allows to set an RSS_LIMIT right inside the request. It would allow to retain the original behavior even more. Still needs an upgrade guide entry IMHO so that users can reconfigure their clients

File: core/src/main/java/hudson/model/Job.java (hunk 1 above)
Comment (daniel-beck) [reply core/src/main/java/hudson/model/Job.java:139]: Why, given it's always the effective limit?

**Comment by daniel-beck** [comment]
I don't think we need to specifically support serving very old builds from the start. Let's wait for people to actually request it.

The problem with limiting to N builds however is that very frequently run jobs will mean that feed readers will have no chance to list them all even with relatively frequent polling. A reasonable compromise could be to return all builds newer than X hours (e.g. 24), and if that's fewer than Y (e.g. 20), return the newest Y builds. That would fit the periodic polling model of feed readers without requiring loading old builds even if they're kept around on disk.

**Comment by basil** [comment]
There hasn't been a response from the original author since this PR was opened 5 months ago, so I am labeling this PR as `stalled`. In the absence of a response I plan to transition it to `proposed-for-close` in approximately one month and close it approximately one week after that.

**Pull request closed without merging**